docs/getting-started/cluster-class/provision.md

```diff
@@ -0,0 +1,140 @@
+
+:::note
+This guide uses the [examples repository](https://github.com/rancher-sandbox/rancher-turtles-fleet-example/tree/templates).
+:::
+
+## Providers guide
+
+<Tabs>
+  <TabItem value="azure-aks" label="CAPZ AKS" default>
+  ## Prerequisites
+    - Rancher Manager cluster with Rancher Turtles installed
+    - Configure cloud credentials for Azure in Rancher: `Cluster Management` > `Cloud Credentials`.
+        - Keep the name you assign to the new set of credentials.
+    - Install the [CAPI Infrastructure Provider for Azure](https://github.com/kubernetes-sigs/cluster-api-provider-azure/) using the [`CAPIProvider` resource](../../tasks/capi-operator/basic_cluster_api_provider_installation.md).
+        ```yaml
+        apiVersion: v1
+        kind: Namespace
+        metadata:
+        name: capz-system
+        ---
+        apiVersion: turtles-capi.cattle.io/v1alpha1
+        kind: CAPIProvider
+        metadata:
+        name: azure
+        namespace: capz-system
+        spec:
+        type: infrastructure
+        name: azure
+        credentials:
+            rancherCloudCredential: <rancher-credential-name> # Rancher credentials secret for Azure
+        configSecret:
+            name: azure-variables
+        variables:
+            CLUSTER_TOPOLOGY: "true"
+            EXP_CLUSTER_RESOURCE_SET: "true"
+            EXP_MACHINE_POOL: "true"
+            EXP_AKS_RESOURCE_HEALTH: "true"
+        ```
+    - Install the [Kubeadm bootstrap provider](https://cluster-api.sigs.k8s.io/tasks/bootstrap/kubeadm-bootstrap/) using the [`CAPIProvider` resource](../../tasks/capi-operator/basic_cluster_api_provider_installation.md). 
+        ```yaml
+        ---
+        apiVersion: v1
+        kind: Namespace
+        metadata:
+        name: capi-kubeadm-bootstrap-system
+        ---
+        apiVersion: turtles-capi.cattle.io/v1alpha1
+        kind: CAPIProvider
+        metadata:
+        name: kubeadm-bootstrap
+        namespace: capi-kubeadm-bootstrap-system
+        spec:
+        name: kubeadm
+        type: bootstrap
+        configSecret:
+            name: variables
+        ---
+        apiVersion: v1
+        kind: Namespace
+        metadata:
+        name: capi-kubeadm-control-plane-system
+        ---
+        apiVersion: turtles-capi.cattle.io/v1alpha1
+        kind: CAPIProvider
+        metadata:
+        name: kubeadm-control-plane
+        namespace: capi-kubeadm-control-plane-system
+        spec:
+        name: kubeadm
+        type: controlPlane
+        configSecret:
+            name: variables
```

Comment:
We don’t need `kubeadm` provider to deploy AKS cluster

Reply:
Yes, that is correct. This is needed to satisfy CAPI's webhooks that require a valid reference to a bootstrap object. Whatever is referenced here won't be used at any point in provisioning. For now I think I'll change it to RKE2, as it will simplify things for Turtles' users but I'll be taking a look at how this is implemented upstream and whether we can remove this hard requirement because this will happen with any other provider that supports managed clusters (namely aws, gcp).